File: Act 5 - RLC Compressor/linked_list.cpp
#include <iostream>

class linkedList;

class node{
    private:
        node* next;
        unsigned char data;
    public:
        node(){ this->next = nullptr; };
        node(unsigned char data){ 
            this->next = nullptr; 
            this->data = data;
        };
    friend class linkedList;
};

class linkedList{
    private:
        node* header;
    public:
        linkedList(){
            this->header = nullptr;
        }
        ~linkedList(){
            node* current = this->header;
            while(current){
                node* next = current->next;
                delete current;
                current = next;
            }
        }
        size_t size();
        void insert(unsigned char);
    unsigned char &operator[](const int&);
};

unsigned char &linkedList::operator[](const int &position) {
    node *aux = this->header;

    for (size_t i = 0; i < position; i++){
        aux = aux->next;
    }

    return aux->data;
}

size_t linkedList::size(){

    node* aux = this->header;
    size_t counter = 0;

    while(aux){
        counter++;
        aux = aux->next;
    }

    return counter;
}

void linkedList::insert(unsigned char data){

    node* value = new node(data);
    node* current = this->header;

    if(!this->header){
        this->header = value;
    }else{
        while(current->next){
            current = current->next;
        }
        current->next = value;
    }

}

File: Act 1 - Linked List/headers/linked_list.py

class contact:
    def __init__(self, id: int, name: str, address: str, phone: str, email: str) -> None:
        self.id = id
        self.name = name
        self.address = address
        self.phone = phone
        self.email = email

    def __str__(self) -> str:
        return f"ID: #{self.id} - {self.name}, {self.address}, {self.phone[:2]}-{self.phone[2:6]}-{self.phone[6:]}, {self.email}"

class node:
    def __init__(self, data: contact = None) -> None:
        self.data = data
        self.next = None

class linked_list:
    def __init__(self, ) -> None:
        self.header = node()

    def empty(self) -> bool:
        if self.header.next != None:
            return False
        return True

    def insert(self, data: contact) -> None:

        new_node: node = node(data)
        aux: node = self.header

        while aux.next != None:
            aux = aux.next

        aux.next = new_node

    def erase(self, name: str) -> None:
        
        current: node = self.header
        last: node = None

        while not self.empty():
            last = current
            current = current.next

            if current.data.name == name:
                last.next = current.next
                return

    def erase_all(self) -> None:
        self.header = node()

    def find(self, name: str) -> dict:
        
        current: node = self.header
        status: bool = False
        data: contact = None

        form: dict = {}

        while not self.empty():
            current = current.next
            if current.data.name == name:
                data = current.data
                status = True
                break

        form['status'] = status
        form['data'] = data
    
        return form

    def show_contacts(self) -> None:
        
        current: node = self.header

        while current.next != None:
            current = current.next
            print(current.data)